### PR TITLE
fix(tickformatter): add timeZone to tickFormatter

### LINKED ISSUE #427: Not passing the timeZone on tickFormat
**Describe the bug**
When using a different time zone, the tickFormatter needs to use the same timezone to correctly format the ticks.
Right now we are not passing the timeZone to the tickFormat and our exposed `niceTimeFormatter` we are using the default timezone, causing some issues to everyone is using that formatter

**To Reproduce**
1. Create a chart with a different timezone
2. Use the `niceTimeFormatter`
3. You will notice that tick will not be correctly formatted

**Expected behavior**
The `niceTimeFormatter` should use the `timezone` parameter passed to the BarSeries.

**Additional context**
n/a niceTimeFormatter**Errors in browser console**
n/a

**Kibana Cross Issues**
n/a

**Checklist**
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- ~[ ] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present~

### PR DESCRIPTION
**Pull request by markov00**
## Summary

to correctly handle a the tick formatting on time based series, we need to share the timeZone
parameter to the tickFormatter. When the tickFormatter is an external function, the user can
manually configure that parameter, but for the niceTimeFormatter we don't have that option already.
Instead of adding a breaking change I've added an optional parameter to the tickFormat props to
share the configured timeZone parameter

fix #427

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- ~[ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [ ] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by markov00** [comment]
I'm expecting some VRT errors because of the use of `local` as timezone

File: src/utils/data/formatters.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { DateTime, Interval } from 'luxon';
+import { TickFormatter, TickFormatterOptions } from '../../chart_types/xy_chart/utils/specs';
 
-type TimeFormatter = (value: number) => string;
-
-export function timeFormatter(format: string): TimeFormatter {
-  return (value: number): string => {
-    return DateTime.fromMillis(value).toFormat(format);
+export function timeFormatter(format: string): TickFormatter {
+  return (value: number, options?: Partial<TickFormatterOptions>): string => {
```
Comment (nickofthyme) [inline src/utils/data/formatters.ts]: This is already a `Partial` type

https://github.com/elastic/elastic-charts/blob/8015f5eef821c9daed4a60aa8212dea5d5130170/src/chart_types/xy_chart/utils/specs.ts#L303-L305

File: src/chart_types/xy_chart/utils/specs.ts [hunk 2]
```diff
@@ -300,7 +300,10 @@ export interface AxisSpec {
   integersOnly?: boolean;
 }
 
-export type TickFormatter = (value: any) => string;
+export type TickFormatterOptions = Partial<{
+  timeZone: string;
+}>;
```
Comment (nickofthyme) [inline src/chart_types/xy_chart/utils/specs.ts]: I think type definitions should set type as is. The advanced types like `Partial`, `Omit`, `Pick` are used with defined types to suit a given need, say I only need one property on the type so I `Pick` that property.

```ts
export interface TickFormatterOptions {
  timeZone?: string;
};
```

**Comment by nickofthyme** [comment]
@rshen91 with the addition of `moment`, it appears to fail gracefully by falling back to the local/browser time zone.

image

**Comment by markov00** [comment]
> @rshen91 with the addition of `moment`, it appears to fail gracefully by falling back to the local/browser time zone.
> 
> image

@nickofthyme @rshen91 do we want to catch that moment exception and throw just a more "nicer" console.warn ?

File: src/chart_types/xy_chart/utils/specs.ts (hunk 2 above)
Comment (markov00) [reply src/chart_types/xy_chart/utils/specs.ts]: solved in https://github.com/elastic/elastic-charts/pull/430/commits/7cca9e814d3ebbe8362342b51936e177cc45a342

**Comment by nickofthyme** [comment]
> @nickofthyme @rshen91 do we want to catch that moment exception and throw just a more "nicer" console.warn ?

No I don't think that is needed. Wylie pointed out our lack of error catching, which he's right. I think it's a bigger issue to handle all bad inputs to chart "nicely" in both dev and production.

One thing we could add for this is an enum but that would limit the flexibility of kibana to just pass a string. I think `moment.tz` is resilient enough to handle most edge cases 💪 .

File: src/utils/data/formatters.ts (hunk 1 above)
Comment (markov00) [reply src/utils/data/formatters.ts]: solved in https://github.com/elastic/elastic-charts/pull/430/commits/7cca9e814d3ebbe8362342b51936e177cc45a342

**Pull request merged**